Source/Protector.swift

```diff
@@ -40,36 +70,46 @@ final class UnfairLock {
         unfairLock.deallocate()
     }
 
-    private func lock() {
+    fileprivate func lock() {
         os_unfair_lock_lock(unfairLock)
     }
 
-    private func unlock() {
+    fileprivate func unlock() {
         os_unfair_lock_unlock(unfairLock)
     }
+}
+#endif
 
-    /// Executes a closure returning a value while acquiring the lock.
-    ///
-    /// - Parameter closure: The closure to run.
-    ///
-    /// - Returns:           The value the closure generated.
-    func around<T>(_ closure: () -> T) -> T {
-        lock(); defer { unlock() }
-        return closure()
+/// A `pthread_mutex_t` wrapper.
+final class MutexLock: Lock {
+    private var mutex: pthread_mutex_t
+    
+    init() {
```

Comment:
As a side note, I think the handling of the mutex here needs to be directly allocated, so avoid similar issues seen with `os_unfair_lock`, as mentioned on [the Swift Forums](https://forums.swift.org/t/atomic-property-wrapper-for-standard-library/30468/18).

Reply:
Looks like `os_unfair_lock` is an UnsafeMutablePointer, but `pthread_mutex_t` is not, so I can’t call `.allocate(1)` like you do for the former.

Reply:
random drive-by comment: @jshier is right, you will need to allocate the memory for `phread_mutex_t` on the heap manually. Swift doesn't guarantee that `&mutex` is always the same pointer so this lock implementation might randomly fail. Also, it fails to check the `pthread_*` return codes. See SwiftNIO's [lock implementation](https://github.com/apple/swift-nio/blob/master/Sources/NIOConcurrencyHelpers/lock.swift#L34) for something that should work.

Reply:
Thanks @weissi! I’ve updated the implementation to allocate manually and check for errors.